packages/internal/dex/sdk/src/exchange.ts

```diff
@@ -143,6 +145,133 @@ export class Exchange {
+      approval,
+    };
+  }
+
+  private getWrapTransaction(
+    fromAddress: string,
+    tokenAmount: CoinAmount<ERC20>,
+    wimxInterface: ethers.utils.Interface,
+    gasPrice: CoinAmount<Native> | null,
+  ) : WrapUnwrapTransactionDetails {
+    const calldata = wimxInterface.encodeFunctionData('deposit');
+    // 45038 is the upper bound of the gas estimate for the deposit function.
+    // This upper bound is reached when a user is wrapping for the first time (cold + unused storage slot)
+    // The lower bound is 27,938, so the difference is 17,100 gas. If transactions are 10 gwei IMX,
+    // this is about 0.000171 IMX difference. In order to avoid an extra remote call, we just use the upper bound instead.
+    const gasEstimate = ethers.BigNumber.from('45038');
+
+    const gasFeeEstimate = gasPrice ? toPublicAmount(calculateGasFee(gasPrice, gasEstimate)) : null;
+    // Get the transaction details. This is calling `deposit` or `withdraw` on the WETH/WIMX contract.
+    const transactionDetails: TransactionDetails = {
+      transaction: {
+        data: calldata,
+        to: this.wrappedNativeToken.address, // wrapping and unwrapping is done on the WETH/WIMX contract itself
+        value: ethers.BigNumber.from(tokenAmount.value), // Wrapping involves sending the native asset as TX.value
+        from: fromAddress,
+      },
+      gasFeeEstimate,
+    };
+
+    return {
+      transaction: transactionDetails,
+      approval: null,
+    };
+  }
+
+  private async getUnsignedWrapUnwrapTx(
+    tokenIn: Coin,
+    otherToken: Coin,
+    amountSpecified: CoinAmount<Coin>,
+    fromAddress: string,
+    gasPrice: CoinAmount<Native> | null,
+  ): Promise<TransactionResponse> {
+    const isWrap = tokenIn === this.nativeToken;
+    const otherTokenCoinAmount = {
+      token: otherToken,
+      value: amountSpecified.value,
+    };
+
+    // The quote is always 1:1 (0 {price impact, slippage}) when wrapping/unwrapping.
+    const quote: Quote = {
+      amount: toPublicAmount(otherTokenCoinAmount),
+      amountWithMaxSlippage: toPublicAmount(otherTokenCoinAmount),
+      slippage: 0,
+      fees: [],
+    };
+
+    const wimxInterface = WIMX__factory.createInterface();
+    let transactionDetails;
+
+    if (isWrap) {
+      transactionDetails = this.getWrapTransaction(
+        fromAddress,
+        otherTokenCoinAmount as CoinAmount<ERC20>,
+        wimxInterface,
+        gasPrice,
+      );
+    } else {
+      transactionDetails = await this.getUnwrapTransaction(
+        fromAddress,
+        otherTokenCoinAmount as CoinAmount<ERC20>,
+        wimxInterface,
+        gasPrice,
+      );
+    }
+
+    return { quote, approval: transactionDetails.approval, swap: transactionDetails.transaction };
+  }
+
+  private isWrapOrUnwrap(tokenIn: Coin, tokenOut: Coin): boolean {
+    return (
+      (
+        tokenIn === this.nativeToken && (
+          tokenOut.type === 'erc20' && tokenOut.address === this.wrappedNativeToken.address
```

Comment:
Knowing whether a given token is the wrapped native seems like the responsibility of the NativeTokenService. How about we add a method there to do this check? It would make this less verbose and more readable